### PR TITLE
Adds support for custom where bounds

### PR DESCRIPTION
**Pull request by bkchr**
The user can now specify a custom where bound when using the derive
macros:

- `#[codec(encode_bound(T: Encode))]` for `Encode`
- `#[codec(decode_bound(T: Encode))]` for `Decode`

If nothing is specified (`encode_bound()`) the where bounds will be empty.

### CODE REVIEW

**Comment by apopiak** [comment]
This sounds like a feature additoin -> minor version increase to me?

File: derive/Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "parity-scale-codec-derive"
 description = "Serialization and deserialization derive macro for Parity SCALE Codec"
-version = "2.0.1"
+version = "2.0.2"
```
Comment (gavofyork) [inline derive/Cargo.toml:4]:
```suggestion
version = "2.1.0"
```

File: Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "parity-scale-codec"
 description = "SCALE - Simple Concatenating Aggregated Little Endians"
-version = "2.0.1"
+version = "2.0.2"
```
Comment (gavofyork) [inline Cargo.toml:4]:
```suggestion
version = "2.1.0"
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,13 @@ All notable changes to this crate are documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this crate adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.0.2] - 2021-04-06
```
Comment (gavofyork) [inline CHANGELOG.md:7]:
```suggestion
## [2.1.0] - 2021-04-06
```

**Review by gavofyork — APPROVED** [review]

**Pull request merged**